spec/fixtures/custom_nginx.template

```diff
@@ -312,60 +312,156 @@ http {
+                local mu = require('spec.fixtures.mock_upstream')
+
+                return mu.send_json_response({
+                    args    = ngx.req.get_uri_args(),
+                    headers = ngx.req.get_headers(),
+                    origin  = ngx.var.remote_addr,
+                    url     = mu.get_request_url(),
+                })
+            }
+        }
+        location = /post {
+            access_by_lua_block {
+                local mu = require('spec.fixtures.mock_upstream')
+                return mu.filter_access_by_method("POST")
+            }
+            content_by_lua_block {
+                local mu                 = require('spec.fixtures.mock_upstream')
+                local headers            = ngx.req.get_headers()
+                local data, form, params = mu.parse_post_data(headers)
+                return mu.send_json_response({
+                    args    = ngx.req.get_uri_args(),
+                    data    = data,
+                    form    = form,
+                    headers = headers,
+                    origin  = ngx.var.remote_addr,
+                    params  = params,
+                    url     = mu.get_request_url(),
+                })
+            }
+        }
+        location ~ "^/basic-auth$" {
+            return 301 /basic-auth/user/passwd;
+        }
+        location ~ "^/basic-auth/(?<username>[a-zA-Z0-9_]+)/(?<password>[.]+)$" {
+            access_by_lua_block {
+                local mu = require('spec.fixtures.mock_upstream')
+                return mu.filter_access_by_basic_auth(ngx.var.username,
+                                                      ngx.var.password)
+            }
+            content_by_lua_block {
+                local mu = require('spec.fixtures.mock_upstream')
+                return mu.send_json_response({
+                  authenticated = true,
+                  user          = ngx.var.username,
+                })
+            }
+        }
+        location ~ "^/(request|anything)$" {
+            content_by_lua_block {
+                local mu                 = require('spec.fixtures.mock_upstream')
+                local headers            = ngx.req.get_headers()
+                local data, form, params = mu.parse_post_data(headers)
+
+                return mu.send_json_response({
+                    args    = ngx.req.get_uri_args(),
+                    data    = data,
+                    form    = form,
+                    headers = headers,
+                    method  = ngx.var.request_method,
+                    origin  = ngx.var.remote_addr,
+                    params  = params,
+                    url     = mu.get_request_url(),
+                })
+            }
+        }
+        location = "/delay" {
+            return 301 /delay/3;
+        }
+        location ~ "^/delay/(?<delay_seconds>\d{3})$" {
+            content_by_lua_block {
+                local delay_seconds = tonumber(ngx.var.delay_seconds)
```

Comment:
It seems to me like httpbin.org's `/delay` only accepts `GET` requests? Is there a test that requires it to support another method on our side? If not, this reduces the size of this handler which is nice (less code is better!)